sdk/storage/src/clients/storage_client.rs

```diff
@@ -34,6 +34,92 @@ pub enum StorageCredentials {
     Anonymous,
 }
 
+impl StorageCredentials {
+    /// Create an Access Key based credential
+    ///
+    /// When you create a storage account, Azure generates two 512-bit storage
+    /// account access keys for that account. These keys can be used to
+    /// authorize access to data in your storage account via Shared Key
+    /// authorization.
+    ///
+    /// ref: https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage
+    pub fn access_key<A, K>(account: A, key: K) -> Self
+    where
+        A: Into<String>,
+        K: Into<String>,
+    {
+        Self::Key(account.into(), key.into())
+    }
+
+    /// Create a Shared Access Signature (SAS) token based credential
+    ///
+    /// SAS tokens provide delegated access to resources in a storage account
+    /// with granular control over how the client can access data in the
+    /// account.
+    ///
+    /// ref: https://docs.microsoft.com/azure/storage/common/storage-sas-overview
```

Comment:
This doc and the linked docs don't really reference the fact that `token` is expected to be a sas query string. Can we make that clear?